client/state/themes/actions.js

```diff
@@ -418,6 +421,51 @@ export function clearActivated( siteId ) {
 }
 
 /**
+ * Triggers a network request to activate a specific wpcom theme on a given Jetpack site.
+ * First step of the process is the installation of the theme on Jetpack site.
+ * Second step is the actuall activation.
+ *
+ * @param  {Number|String} siteId    Site ID
+ * @param  {String}        themeId   Theme ID, this should be standard id without -wpcom suffix.
+ * @param  {String}        source    The source that is reuquesting theme activation, e.g. 'showcase'
+ * @param  {Boolean}       purchased Whether the theme has been purchased prior to activation
+ * @return {Function}                Action thunk
+ */
+export function activateWpcomThemeOnJetpack( siteId, themeId, source = 'unknown', purchased = false ) {
+	//Add -wpcom suffix. This suffix tells the endpoint that we want to
+	//install WordPress.com theme. Without the suffix endpoint would look
+	//for theme in .org
+	const wpcomThemeId = themeId + '-wpcom';
```

Comment:
I understand why `wpcomThemeId` is named so, but it actually is the _Jetpack_ theme id, because on wpcom the suffix is not necessary. Maybe we should call it something like `suffixedThemeId`?